src/OmniSharp.Roslyn.CSharp/Services/Diagnostics/CodeCheckService.cs

```diff
@@ -13,22 +13,47 @@ namespace OmniSharp.Roslyn.CSharp.Services.Diagnostics
     [OmniSharpHandler(OmniSharpEndpoints.CodeCheck, LanguageNames.CSharp)]
     public class CodeCheckService : IRequestHandler<CodeCheckRequest, QuickFixResponse>
     {
-        private OmniSharpWorkspace _workspace;
+        private readonly OmniSharpWorkspace _workspace;
+        private readonly RoslynAnalyzerService _roslynAnalyzer;
+        private readonly ILogger<CodeCheckService> _logger;
 
         [ImportingConstructor]
-        public CodeCheckService(OmniSharpWorkspace workspace)
+        public CodeCheckService(OmniSharpWorkspace workspace, RoslynAnalyzerService roslynAnalyzer, ILoggerFactory loggerFactory)
         {
             _workspace = workspace;
+            _roslynAnalyzer = roslynAnalyzer;
+            _logger = loggerFactory.CreateLogger<CodeCheckService>();
         }
 
         public async Task<QuickFixResponse> Handle(CodeCheckRequest request)
         {
-            var documents = request.FileName != null
-                ? _workspace.GetDocuments(request.FileName)
-                : _workspace.CurrentSolution.Projects.SelectMany(project => project.Documents);
+            var projectsForAnalysis = !string.IsNullOrEmpty(request.FileName)
```

Comment:
There is a different service that duplicates this logic for push diagnostics.

Can you also update https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp.Roslyn.CSharp/Workers/Diagnostics/CSharpDiagnosticService.cs#L143 to have the same behavior?